src/context/MakeupContext.jsx

```diff
@@ -0,0 +1,33 @@
+import { useContext, useEffect } from 'react';
+import { useState } from 'react';
+import { createContext } from 'react';
+import { getMakeups } from '../services/makeups';
+
+const MakeupContext = createContext();
+
+export function MakeupProvider({ children }) {
+  const [makeups, setMakeups] = useState([]);
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      const data = await getMakeups();
+      const newData = [...new Set(data)];
+      setMakeups(newData);
+    };
+    fetchData();
+    setLoading(false);
+  }, [setLoading]);
+  if (loading) return <h1>im loading</h1>;
```

Comment:
We should just return the loading state from the function, and keep the conditional inside of the component.